test/server/test-auth.js

```diff
@@ -1,13 +1,53 @@
+import sinon from 'sinon';
 import { store } from 'App.js';
 import { AuthStates } from 'redux/actions';
 // import logger from 'logger/logger.js';
 import { login, logout } from 'server/user-auth';
-
+import * as tokenManager from 'server/core/tokenmanager';
+import * as connection from 'server/core/connection';
 import { user38257001 as testUser } from './core/users';
+import { testToken1 as testToken } from './core/tokens';
+
+const sinonChai = require('sinon-chai');
+const chai = require('chai');
 
-const expect = require('chai').expect;
+chai.use(sinonChai);
+const expect = chai.expect;
 
 export default describe('Server API: Auth', () => {
+  before(() => {
+    sinon.stub(tokenManager, 'saveToken');
+    sinon.stub(tokenManager, 'removeToken');
+    sinon.stub(connection, 'post').callsFake((endpoint, data, onSuccess, onError) => {
+      if (endpoint === '/auth') {
+        onSuccess({
+          token: testToken.token,
+        });
+      } else {
+        onError(new Error('Test: Invalid post endpoint'));
+      }
+    });
+    sinon.stub(connection, 'del').callsFake((endpoint, data, onSuccess, onError) => {
+      if (endpoint === '/auth') {
+        onSuccess();
+      } else {
+        onError(new Error('Test: Invalid del endpoint'));
+      }
+    });
+  });
+
+  afterEach(() => {
+    tokenManager.saveToken.resetHistory();
+    tokenManager.removeToken.resetHistory();
+    connection.post.resetHistory();
+  });
+
+  after(() => {
+    tokenManager.saveToken.restore();
```

Comment:
This is strictly necessary. The lifetime of the mocking outlives the tests. I discovered that when, without this line, the `test-core-tokenmanager` tests failed.